### PR TITLE
th/dns-stub-resolv-conf

### PR DESCRIPTION
**Pull request by thom311**
We write `/var/run/NetworkManager/resolv.conf` with the same content that we push to the rc-manager (according to `main.rc-manager` setting). In combination with a caching DNS plugin, the name servers there are either `127.0.0.1` or `.127.0.0.53`.

Write another file `/var/run/NetworkManager/stub-resolv.conf` which contains the original nameservers.

### CODE REVIEW

**Comment by thom311** [comment]
TODO: rename "stub-resolv.conf" to "no-stub-resolv.conf"

File: man/NetworkManager.conf.xml [hunk 1]
```diff
@@ -331,15 +331,23 @@ no-auto-default=*
...
         modify resolv.conf. This implies
         <literal>rc-manager</literal>&nbsp;<literal>unmanaged</literal></para>
+
+        <para>Note that the plugins <literal>dnsmasq</literal>, <literal>systemd-resolved</literal>
+        and <literal>unbound</literal> are caching local nameservers.
+        Hence, when NetworkManager writes <filename>&nmrundir;/resolv.conf</filename>
+        and <filename>/etc/resolv.conf</filename> (according to <literal>rc-manager</literal>
+        setting below), the name server there will be localhost only.
+        NetworkManager also writes a file <filename>&nmrundir;/no-stub-resolv.conf</filename>
+        that contains the original name servers, which are pushed to the DNS plugin.</para>
```
Comment (bengal) [inline man/NetworkManager.conf.xml]: "that contains the original name servers pushed to the DNS plugin.</para>" ?

**Comment by bengal** [comment]
New 2 commits look good to me

**Comment by thom311** [comment]
merged

**Pull request closed without merging**